### ISSUE TITLE
inf frames of shapre 2304x1536 at 2.00fps

### ISSUE THREAD

**Issue by andjimenezuf** [ultralytics/ultralytics#19510]
### Search before asking

- [ ] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I'm currently learning object detection using YOLOv11n and running into performance issues on my setup. I have a Jetson AGX Orin from Advantech AIR 030 and I’m using a Logitech 2920 camera. I'm running CUDA 12.2, Torch 2.3, and Tensorvision 0.18, and I followed the documentation closely. However, when I run YOLO prediction on both the yolov11n.pt model and the yolo11n.engine file, I only get around 2 FPS at a resolution of 2304x1536

Here’s what I’ve tried so far:
- Lowering the input image size.
- Reducing the camera resolution.
- Setting the power mode to maximum.
- Enabling Jetson clocks for performance boost.

when using DeepStream, I’m able to achieve much better frame rates.

Does anyone have suggestions on additional tweaks that might help improve my FPS for real-time object detection? Feel free to ask for more information if needed.

Thanks in advance for your help!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @andjimenezuf, thank you for reaching out and sharing your experience with Ultralytics 🚀! We’re here to help. Here's some guidance and resources to get started:

For performance issues like this, it's helpful to provide a bit more context and a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) 🐛 if applicable. This allows us to better understand your setup and debug effectively.

### Suggestions:

1. Please verify that you’re using the latest `ultralytics` package and all dependencies are updated. You can upgrade using the following command:
   ```bash
   pip install -U ultralytics
   ```
   Make sure to operate in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

2. If you are running inference on Jetson devices, ensure you are leveraging TensorRT optimizations effectively. This might include fine-tuning engine generation settings to match your specific hardware.

3. Kindly share more details:
   - Exact commands you’re running for predictions.
   - Any relevant logs or warnings during runtime.
   - FPS benchmarks you’ve achieved with DeepStream for comparison.

4. Performance can vary significantly depending on the environment and hardware. If possible, try running your setup in any of these tested environments:
   - **Notebooks** with free GPU:  
     <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
   - **Docker Image**: Prebuilt environments for seamless integration, see: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

5. Additionally, you may find optimization tips in the [Docs](https://docs.ultralytics.com/guides/model-training-tips/). For camera integrations, it’s worth exploring FPS limits imposed by the camera interface or processing pipeline.

### Resources:
- Real-time advice ➡️ [Discord](https://discord.com/invite/ultralytics) 🎧  
- Community support ➡️ [Discourse](https://community.ultralytics.com/) or [Subreddit](https://reddit.com/r/Ultralytics)  

An Ultralytics engineer will review your case soon to provide further insights! 😊 Thank you for your patience and for being a part of the Ultralytics community.

**Comment by andjimenezuf** [comment]
> What's your code
I didn't modify or write any custom code—I simply ran the commands listed in the "Quick Start Guide: NVIDIA Jetson with Ultralytics YOLOv11." For training, I executed:

`yolo detect train data=data.yaml model=yolo11n.pt epochs=100`

For prediction, I ran:

`yolo detect predict model=runs/detect/trainX/weights/best.pt source=0 conf=0.25 show=true`

I also exported the best model from `.pt` to `.engine`, but I'm still getting around 2 FPS. Let me know if you need any more details!

**Comment by Y-T-G** [comment]
What's the prediction log with latency?

**Comment by Y-T-G** [comment]
Did you use half=True to export to TensorRT?

**Comment by Y-T-G** [comment]
The PyTorch model will always be slower than DeepStream. You should be using the .engine model. Export with `half=True` and `nms=True`

**Comment by andjimenezuf** [comment]
> What's the prediction log with latency?

ubuntu@localhost:~/Datasets/mask_detection_project$ yolo detect predict model=runs/detect/train3/weights/best.engine half=true show=true source=0 conf=0.25
Ultralytics 8.3.80 🚀 Python-3.10.12 torch-2.3.0 CUDA:0 (Orin, 62842MiB)
Loading runs/detect/train3/weights/best.engine for TensorRT inference...
[03/06/2025-04:07:21] [TRT] [I] Loaded engine size: 5 MiB
[03/06/2025-04:07:21] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in engine deserialization: CPU +0, GPU +2, now: CPU 0, GPU 2 (MiB)
[03/06/2025-04:07:21] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +4, now: CPU 0, GPU 6 (MiB)

[ WARN:0@4.453] global cap_gstreamer.cpp:1777 open OpenCV | GStreamer warning: Cannot query video position: status=0, value=-1, duration=-1
1/1: 0... Success ✅ (inf frames of shape 2304x1536 at 2.00 FPS)

[03/06/2025-04:07:24] [TRT] [E] 3: [executionContext.cpp::validateInputBindings::2103] Error Code 3: API Usage Error (Parameter check failed at: runtime/api/executionContext.cpp::validateInputBindings::2103, condition: profileMaxDims.d[i] >= dimensions.d[i]. Supplied binding dimension [1,3,640,640] for bindings[0] exceed min ~ max range at index 2, maximum dimension in profile is 416, minimum dimension in profile is 32, but supplied dimension is 640.
)
[03/06/2025-04:07:24] [TRT] [E] 3: [executionContext.cpp::validateInputBindings::2103] Error Code 3: API Usage Error (Parameter check failed at: runtime/api/executionContext.cpp::validateInputBindings::2103, condition: profileMaxDims.d[i] >= dimensions.d[i]. Supplied binding dimension [1,3,448,640] for bindings[0] exceed min ~ max range at index 2, maximum dimension in profile is 416, minimum dimension in profile is 32, but supplied dimension is 448.
)
0: 448x640 (no detections), 4.8ms
0: 448x640 (no detections), 8.2ms
0: 448x640 (no detections), 5.9ms
0: 448x640 (no detections), 5.7ms
0: 448x640 (no detections), 5.9ms
0: 448x640 (no detections), 5.4ms
0: 448x640 (no detections), 6.1ms
0: 448x640 (no detections), 14.2ms
0: 448x640 (no detections), 6.4ms
0: 448x640 (no detections), 6.3ms
0: 448x640 (no detections), 13.3ms
0: 448x640 (no detections), 11.5ms
0: 448x640 (no detections), 10.0ms
0: 448x640 (no detections), 12.3ms
0: 448x640 (no detections), 6.1ms
0: 448x640 (no detections), 7.1ms
0: 448x640 (no detections), 6.1ms
0: 448x640 (no detections), 5.4ms
0: 448x640 (no detections), 6.2ms

**Comment by andjimenezuf** [comment]
> Did you use half=True to export to TensorRT?
Yes i tried both

**Comment by Y-T-G** [comment]
It doesn't seem like a model issue. The camera only reports 2 FPS.

**Comment by andjimenezuf** [comment]
> It doesn't seem like a model issue. The camera only reports 2 FPS.
Hmm but when i run a sample deepstream app I get 30 fps. I was also able to use a repository to run the YOLO model on DeepStream and achieved 10 FPS. Additionally, the camera works well with the Cheese app. Do you have any other suggestions regarding parameters I should use when exporting or running inference?

**Comment by Y-T-G** [comment]
You can manually run the camera using Ultralytics.

```python
import cv2

from ultralytics import YOLO

# Load the YOLO11 model
model = YOLO("yolo11n.engine")
cap = cv2.VideoCapture(0)

# Loop through the video frames
while cap.isOpened():
    # Read a frame from the video
    success, frame = cap.read()

    if success:
        results = model(frame)

        # Visualize the results on the frame
        annotated_frame = results[0].plot()

        # Display the annotated frame
        cv2.imshow("YOLO11", annotated_frame)

        # Break the loop if 'q' is pressed
        if cv2.waitKey(1) & 0xFF == ord("q"):
            break
    else:
        # Break the loop if the end of the video is reached
        break

# Release the video capture object and close the display window
cap.release()
cv2.destroyAllWindows()
```

**Comment by andjimenezuf** [comment]
> You can manually run the camera using Ultralytics.
This is what i see when i run that script:
[03/06/2025-06:02:12] [TRT] [I] Loaded engine size: 6 MiB
[03/06/2025-06:02:12] [TRT] [W] Using an engine plan file across different models of devices is not recommended and is likely to affect performance or even cause errors.
[03/06/2025-06:02:12] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in engine deserialization: CPU +6, GPU +0, now: CPU 6, GPU 0 (MiB)
[03/06/2025-06:02:12] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +22, now: CPU 6, GPU 22 (MiB)

0: 640x640 2 persons, 1 backpack, 2 chairs, 2 tvs, 1 book, 35.4ms
Speed: 44.3ms preprocess, 35.4ms inference, 620.3ms postprocess per image at shape (1, 3, 640, 640)

0: 640x640 1 person, 5 chairs, 3 tvs, 1 book, 56.3ms
Speed: 7.5ms preprocess, 56.3ms inference, 11.1ms postprocess per image at shape (1, 3, 640, 640)

0: 640x640 1 person, 4 chairs, 3 tvs, 1 book, 39.4ms
Speed: 13.0ms preprocess, 39.4ms inference, 15.6ms postprocess per image at shape (1, 3, 640, 640)

0: 640x640 1 person, 1 backpack, 4 chairs, 3 tvs, 1 book, 42.1ms
Speed: 16.6ms preprocess, 42.1ms inference, 10.3ms postprocess per image at shape (1, 3, 640, 640)

0: 640x640 1 person, 1 backpack, 4 chairs, 3 tvs, 1 book, 38.0ms
Speed: 13.9ms preprocess, 38.0ms inference, 14.9ms postprocess per image at shape (1, 3, 640, 640)
 Im still getting very low frames.

**Comment by Y-T-G** [comment]
You can export with `nms=True`
